packages/ag-charts-enterprise/src/rangeArea/rangeArea.ts

```diff
@@ -0,0 +1,939 @@
+
+        let lastXValue: any;
+        processedData?.data.forEach(({ keys, datum, values }, datumIdx) => {
+            const xValue = keys[xIndex];
+            const yHighValue = values[yHighIndex];
+            const yLowValue = values[yLowIndex];
+            const yHighTrailingValue = values[yHighTrailingIndex];
+            const yLowTrailingValue = values[yLowTrailingIndex];
+
+            const points = createCoordinates(xValue, yHighValue, yLowValue);
+            points.forEach(({ x, y, size, itemId, yValue }) => {
+                // marker data
+                markerData.push({
+                    index: datumIdx,
+                    series: this,
+                    itemId,
+                    datum,
+                    nodeMidPoint: { x, y },
+                    yHighValue,
+                    yLowValue,
+                    xValue,
+                    xKey,
+                    yLowKey,
+                    yHighKey,
+                    point: { x, y, size },
+                });
+
+                // label data
+                let labelText;
+                if (label.formatter) {
+                    labelText = callbackCache.call(label.formatter, { value: yValue, seriesId: this.id, itemId }) ?? '';
+                } else {
+                    labelText = isNumber(yValue) ? Number(yValue).toFixed(2) : String(yValue);
+                }
+                labelData.push({
+                    series: this,
+                    itemId,
+                    datum,
+                    x,
+                    y: y + (itemId === 'low' ? +10 : -10),
+                    text: labelText,
+                    textAlign: 'center',
+                    textBaseline: itemId === 'low' ? 'top' : 'bottom',
+                });
+            });
+
+            // Handle data in pairs of current and next x and y values
+            const windowX = [lastXValue, xValue];
+            const windowYHigh = [yHighTrailingValue, yHighValue];
+            const windowYLow = [yLowTrailingValue, yLowValue];
+
+            if (windowX.some((v) => v == null)) {
+                lastXValue = xValue;
+                return;
+            }
+            if (windowYHigh.some((v) => v == null) || windowYLow.some((v) => v == null)) {
+                windowYHigh[0] = null;
+                windowYHigh[1] = null;
+                windowYLow[0] = null;
+                windowYLow[1] = null;
+            }
+
+            const trailingCoordinates = createCoordinates(lastXValue, +windowYHigh[0], +windowYLow[0]!);
+            const currentCoordinates = createCoordinates(xValue, +windowYHigh[1], +windowYLow[1]!);
+
+            const highPoints = [trailingCoordinates[0], currentCoordinates[0]];
+            const lowPoints = [trailingCoordinates[1], currentCoordinates[1]];
+
+            // fill data
+            fillHighPoints.push(...highPoints);
+            fillLowPoints.push(...lowPoints);
+
```

Comment:
The processing here seems very similar to `AreaSeries` - is there any commonality we can factor out at this stage?